### ISSUE TITLE
$mdUtil: now is not a function

### ISSUE THREAD

**Issue by poltorak** [angular/material#3762]
I got error on my project: Property 'now' of object #<Object> is not a function

I found out that it happens only on devices with Android 4.0 - 4.2. I also found out that it happens [here](https://github.com/angular/material/blob/master/src/components/sticky/sticky.js#L289).

Problem is that I can't reproduce it. I can't tell more details because that error occurs only on production where all js files are uglified.

Can someone help or has the same problem?

EDIT:
Method ```now``` from $mdUtil looks like:
```javascript
now: window.performance ?
      angular.bind(window.performance, window.performance.now) :
      Date.now,
```
and on caniuse.com I found that neither window.performance.now and Date.now are not supported in Android 4.1 - 4.3

EDIT 2:
Simple solution is to check if window.performance and Date.now is available for example
```javascript
if (window.performance) {
    angular.bind(window.performance, window.performance.now);
    } else if (Date.now) {
        Date.now;
    } else {
        function() {
            return new Date().getTime();
        };
    }

**Comment by yoav-zibin** [comment]
It still happens to me, even with the fix in 53c1f09 , i.e., the code is:
now: window.performance ?
      angular.bind(window.performance, window.performance.now) : Date.now || function() {
      return new Date().getTime();
    }

Error I get:
TypeError: Property 'now' of object #<Object> is not a function


The bug is that window.performance on some systems (android 4.2) doesn't have .now method (but window.performance object does exist).

The solution is to add "&& window.performance.now" like this:
now: window.performance && window.performance.now ?
      angular.bind(window.performance, window.performance.now) : Date.now || function() {
      return new Date().getTime();
    }

**Comment by yoav-zibin** [comment]
If you want to reproduce it, the issue still happens in this system:
{
  "appName": "Netscape",
  "appCodeName": "Mozilla",
  "appVersion": "5.0 (Linux; U; Android 4.2.2; vi-vn; XoLo A500S Build/JDQ39) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30",
  "platform": "Linux armv7l",
  "product": "Gecko",
  "userAgent": "Mozilla/5.0 (Linux; U; Android 4.2.2; vi-vn; XoLo A500S Build/JDQ39) AppleWebKit/534.30 (KHTML, like Gecko) Version/4.0 Mobile Safari/534.30",
  "vendor": "Google Inc."
}

**Comment by gimox** [comment]
+1 same problem with android 4.0 stock browser

**Comment by dinebennett** [comment]
Bugsnag is also reporting this error for me in the following versions of Android browser:
4.0.4, 4.1.1, 4.1.2, 4.2.0, 4.2.1, 4.2.2, 4.4.2, 4.4.4, 5.1.0, 6.1.0

We are using angular material 1.1.1

**Comment by Splaktar** [comment]
@dinebennett that issue was fixed in AngularJS Material 1.1.2 via PR https://github.com/angular/material/pull/9664.
